### PR TITLE
Book: Expanded the planned sections and added built-in module section

### PR DESCRIPTION
**Pull request by antimora**
Expanded the planned sections and added built-in module section

### CODE REVIEW

File: burn-book/src/SUMMARY.md [hunk 1]
```diff
@@ -12,9 +12,23 @@
...
   - [Custom Training Loops]()
   - [Custom Metric]()
   - [Custom Kernels]()
     - [WGPU]()
+  - [WebAssembly]()
+  - [No-Std]()
+- [Contributing]()
+  - [Tenets]()
+  - [Design Philosophy]()
+  - [Architecture]()
```
Comment (nathanielsimard) [inline burn-book/src/SUMMARY.md:33]: I think this should be elsewhere, maybe in another book.

File: burn-book/src/building-blocks/module.md [hunk 2]
```diff
@@ -99,3 +97,33 @@ pub trait ModuleMapper<B: Backend> {
...
+| `Gru`                | `nn.GRU`                                |
+| `LayerNorm`          | `nn.LayerNorm`                          |
+| `Linear`             | `nn.Linear`                             |
+| `Lstm`               | `nn.LSTM`                               |
+| `MaxPool1d`          | `nn.MaxPool1d`                          |
+| `MaxPool2d`          | `nn.MaxPool2d`                          |
+| `MultiHeadAttention` | `nn.MultiheadAttention`                 |
+| `PositionalEncoding` | _No direct equivalent_                  |
+| `TransformerDecoder` | `nn.TransformerDecoder`                 |
+| `TransformerEncoder` | `nn.TransformerEncoder`                 |
```
Comment (nathanielsimard) [inline burn-book/src/building-blocks/module.md:129]: I would remove "a few" since it isn't that small and it will only grow, potentially making this line look odd in the future.

Also, the list isn't complete:

- CrossEntropyLoss
- MSELoss
- LayerNorm
- Embedding
- PositionalEncoding
- Relu

I would also separate them into sub-sections:

- Pooling
- Convolutions
- RNNs
- Transformer
- General
- Loss

At some point, we might create issues that document which modules we want to support in each section.

Finally, the `MSELoss` and `CrossEntropyLoss` don't implement `Module` yet, but we should annotate them even though they don't have state. They were always used inline, but we could put them into a module as is often done with PyTorch.

File: burn-book/src/SUMMARY.md [hunk 3]
```diff
@@ -12,9 +12,23 @@
   - [Tensor](./building-blocks/tensor.md)
   - [Autodiff](./building-blocks/autodiff.md)
   - [Module](./building-blocks/module.md)
+  - [Config]()
+  - [Learner]()
+  - [Metric]()
+  - [Record]()
+  - [Dataset]()
+  - [Optimizer]()
```
[the remark this answers is not in the document]
Comment (antimora) [reply burn-book/src/SUMMARY.md:20]: OK. Moved to advanced section.

File: burn-book/src/SUMMARY.md (hunk 1 above)
Comment (antimora) [reply burn-book/src/SUMMARY.md:33]: OK. Maybe we should add this to CONTRIBUTING.md. I'll remove them.

File: burn-book/src/SUMMARY.md [hunk 4]
```diff
@@ -12,9 +12,23 @@
...
+- [Installation and Setup]()
```
[the remark this answers is not in the document]
Comment (antimora) [reply burn-book/src/SUMMARY.md:21]: OK. I'll move it over there. Since it was empty, I had put it there.

File: burn-book/src/building-blocks/module.md (hunk 2 above)
Comment (antimora) [reply burn-book/src/building-blocks/module.md]: > Finally, the `MSELoss` and `CrossEntropyLoss` don't implement `Module` yet, but we should annotate them even though they don't have state. They were always used inline, but we could put them into a module as is often done with PyTorch.

Filed a ticket for this: https://github.com/burn-rs/burn/issues/736

Rest is updated.

**Review by nathanielsimard — APPROVED** [review]
